app/controllers/immunizations_controller.rb

```diff
@@ -0,0 +1,20 @@
+class ImmunizationsController < ApplicationController
+  def update
+    if admin?
+      @immunization = Immunization.find(params[:id])
+      if @immunization.update(update_immunization_params)
+        render json: { message: 'Immunization updated', status: :updated }
+      else
+        render json: { message: 'Immunization could not be updated. Please try again' }
+      end
+    else
+      render json: { error: 'Not authorized' }, status: :unauthorized
+    end
+  end
+
+  private
+
+  def update_immunization_params
+    params.permit(:id, :vaccination_date)
```

Comment:
You do not really need id here. You do not want anybody to change it, right?

Reply:
@nidalaa Done